### ISSUE TITLE
Support for 16-xf0175ng (8L370EA)

### ISSUE THREAD

**Issue by ghost** [alou-S/omen-fan#7]
Hello, can you add support for 16-xf0175ng (8L370EA) laptop?

**Comment by alou-S** [comment]
Did you attempt to force the current script to run on your laptop?

**Comment by ghost** [comment]
Yep, but it not works

**Comment by alou-S** [comment]
Could you send me your laptop's DSDT table.

```
sudo cp /sys/firmware/acpi/tables/DSDT $HOME/Desktop/DSDT
sudo chown $USER:$USER $HOME/Desktop/DSDT
```

And then upload the DSDT file here

**Comment by ghost** [comment]
[DSDT.zip](https://github.com/alou-S/omen-fan/files/14460691/DSDT.zip)
Here you go, tell if you need any help
Do you have discord or something for communication?

**Comment by alou-S** [comment]
The ERAM fields on your DSDT table practically has no information in it

```
    OperationRegion (ERAM, EmbeddedControl, Zero, 0xFF)
    Field (ERAM, ByteAcc, Lock, Preserve)
    {
        Offset (0x01), 
        Offset (0x02), 
        Offset (0x03), 
        Offset (0x60), 
        SMPR,   8, 
        SMST,   8, 
        SMAD,   8, 
        SMCM,   8, 
        SMD0,   256, 
        BCNT,   8, 
        SMAA,   24, 
        SMBN,   8, 
        Offset (0x8A)
    }
```

You will probably have to figure out the correct offsets yourself by using `ec_probe monitor` from [NBFC](https://github.com/hirschmann/nbfc) on Windows while manually adjusting the fan speed with Omen Gaming Hub.

**Comment by ghost** [comment]
Offsets of what and how do I find that ec_probe?

**Comment by alou-S** [comment]
The thing is that this is actually somewhat complicated.

The `ec_probe` executable should be found in the NBFC install directory.

And you should find the offsets of the Fan Control and probably other things I have a list of my Laptops EC offsets here : <https://github.com/alou-S/omen-fan/blob/main/docs/probes.md>

**Comment by ghost** [comment]
When I try to open executable it closes immidiately

**Comment by alou-S** [comment]
Open it from command prompt :slightly_smiling_face:

**Comment by ghost** [comment]
![image](https://github.com/alou-S/omen-fan/assets/45370592/5fd98a80-2700-434d-9d61-7680b3115516)
I'm getting those numbers, what's next?

**Comment by alou-S** [comment]
You gotta figure out which ones pertain to changing the fan speed when using Omen Gaming Hub to tweak the fan speeds.
Here is a good guide that can help you make sense of it : https://github.com/hirschmann/nbfc/wiki/Probe-the-EC%27s-registers

Also do note that some probes just reflect the fan's current RPM and not the set target RPM by software. So don't be fooled by that.
Good Luck with figuring it out.

**Comment by ghost** [comment]
But it shows like 50 probes and re-rendering all the time, how I supposed to see difference? Does those red ones mean something?

**Comment by alou-S** [comment]
The hex number in green are the addresses on your EC (probes). 
New numbers can appear over time as more of the addresses are modified.

The moving numbers to the right of the probes are their values over time. The red ones are so your eyes can quickly find the ones that change.

Yes there are alot of probes that serve different purposes. Some give temperature, some set power targets, some set fan speed etc. 
Please do check the guide link and my list of offsets link that I have posted before.

**Comment by ghost** [comment]
Should I close Omen spyware when writing probes, or will it not overwrite? I've tried with it both on and off, using all the probes I found that change when I choose another power mode. However, nothing changed and I know it because the energy-saving power mode would limits fan cooling speed.

**Comment by alou-S** [comment]
Wait doesn't the Omen software have a method to manually set the fan speed.

**Comment by ghost** [comment]
It is. but I started with Power mode since there's less values (3)

**Comment by alou-S** [comment]
Uhh just try figuring things out man. It will take a while of trial and error. Just use the resources I have given.

**Comment by ghost** [comment]
But how do I know it worth of wasting time if I can't modify even Power mode, alse there's strings like D8, 0A, D9 (basically any with letter) that I can't write through cmd.
I'm not even sure that I'm overwriting anything...
Updete: okay, those strings was Hexademical numbers, so you need to convert them, and basically I found one of working probes that affect something

**Comment by ghost** [comment]
So, let's start, I will put all the findings here:
0x11: Read only Fan speed?
0x14: Fan speed; 0-64 range; RPM = number * 100; Omen Gaming Hub gives 6000 max and 2500 min. Only left fan for some reason

**Comment by ghost** [comment]
I tried everything and can't find how to turn on right fan through EC, when you write to 0x14 it will only change left fan speed

**Comment by BlueBeret** [comment]
@alou-S I figured out the address to write but there's no file "/sys/kernel/debug/ec/ec0/io"? I am using `ubuntu 24.04  6.8.0-45-generic `.

**Comment by Sharwesh05** [comment]
Hi I am facing the same issue here can you send ur dsdt.aml file to cross check mine @alou-S

**Comment by BlueBeret** [comment]
I don't think it's possible just by probing. You also need to find which register to enable the manual control. The only solution I think is by pry open your laptop and check for EC chip and visit alldatasheet.com.

**Comment by Sharwesh05** [comment]
I understood that, according to my laptop(xd0007AX) 
Left fan is controls the right so by changing right register it only works on right(windows) in Linux it has no response so I would like to know how he found about the bios control and stuff

**Comment by Sharwesh05** [comment]
![IMG20240927135302](https://github.com/user-attachments/assets/15797354-1f94-40dd-b605-fcbd9f7f0f4d)
And I also my lm_sensor doesn't detects my fan speed when it is in always on state(bios settings)
And also I find three errors which are present in my dsdt table

**Comment by KaramelliSundae** [comment]
Well, i have the same issue on HP Victus S0046NT 
0x14 controls right fan only (gpu fan) BUT whatever i try i cannot seem to find left fan's EC location.

@Sharwesh05 Any updates? Perhaps you found the other ec register? I assume we all might be having the same EC chip.

here's my DSDT table but theres basically no information in it: 

[DSDT.zip](https://github.com/user-attachments/files/17955768/DSDT.zip)

**Comment by KaramelliSundae** [comment]
[DSDT decompiled.zip](https://github.com/user-attachments/files/17955808/DSDT.decompiled.zip)
Here's the DSDT decompiled with IASL for convenience. 0x14 seems to control right fan; 0x11 seems to be the read register for fan speed. However, even after trying hundred+ EC registers i cannot find the register for the left fan.

**Comment by Sharwesh05** [comment]
@KaramelliSundae are u saying 0x11 not the left register because I assumed that it is left, and it doesn't affect anything on Linux os when I try to access. and can you respond fast

**Comment by KaramelliSundae** [comment]
Yes you're right, 0x11 is NOT left fan register, it's the register that is
used for reading right fan speed value.

**Comment by Sharwesh05** [comment]
@KaramelliSundae did you try to access it on linux and did it work??

**Comment by KaramelliSundae** [comment]
Yes, editing 0x14 to a value of (0-59) works for me. Only right fan speed
increased.

**Comment by Sharwesh05** [comment]
What os did you try and can you explain the steps because i tried and failed

**Comment by KaramelliSundae** [comment]
I tried on both windows 11 latest and Linux, Fedora 41 with debugging
kernel

**Comment by Sharwesh05** [comment]
@KaramelliSundae is there any other way to contact you

**Comment by KaramelliSundae** [comment]
Yes https://t.me/ctivity
